src/databricks/labs/ucx/framework/parallel.py

```diff
@@ -48,24 +48,22 @@ def _run(self) -> (list[Result], list[Exception]):
             if result is None:
                 continue
             collected.append(result)
-        self._on_finish(given_cnt, len(errors))
+        self._on_finish(given_cnt, len(collected), len(errors))
 
         return collected, errors
 
-    def _on_finish(self, given_cnt, failed_cnt):
+    def _on_finish(self, given_cnt: int, collected_cnt: int, failed_cnt: int):
         since = dt.datetime.now() - self._started
-        failed_pct = 0
-        if failed_cnt > 0:
-            failed_pct = failed_cnt / given_cnt * 100
-        stats = f"{failed_pct:.0f}% ({failed_cnt}/{given_cnt}). Took {since}"
+        success_pct = collected_cnt / given_cnt * 100
+        stats = f"{success_pct:.0f}% results available ({collected_cnt}/{given_cnt}). Took {since}"
         if failed_cnt == given_cnt:
             logger.critical(f"All '{self._name}' tasks failed!!!")
-        elif failed_pct >= self._task_fail_error_pct:
+        elif failed_cnt > 0 and success_pct <= self._task_fail_error_pct:
             logger.error(f"More than half '{self._name}' tasks failed: {stats}")
         elif failed_cnt > 0:
             logger.warning(f"Some '{self._name}' tasks failed: {stats}")
```

Comment:
Not sure why these categories and different log levels depending on how many task failed, even if one task fails it's not good at all.

Reply:
https://github.com/databrickslabs/ucx/issues/406 look at todos in code linking to this issue :)